### PR TITLE
Switch off non-clinical gene warnings when filtering research variants

### LINKED ISSUE #2649: Disable clinical gene list warning on research search
I've just tried to search for the gene DNAH14 in the research list in family '19074-muscle' and got an error message that the gene is not included in the clinical list.  Either I have used the filter incorrectly or the filter is not working.  Has something changed in the filter which means we should change how we search?  Took a screenshot of the error message.  Carried out the same search in the clinical list and got the same error message but when looking for a gene in one of our panels it works no problem.

image

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #2649

**How to test**:
1. On clinical-db stage go to a case that has research variants
2. From research variants page look for a gene that is not in case clinical list (DNAH14?)
3. Make sure you get no warnings

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/blueprints/variants/controllers.py [hunk 1]
```diff
@@ -1000,16 +1000,22 @@ def check_form_gene_symbols(
                 # clinical symbols from gene panels might not be up to date with latest gene names
                 # but their HGNC id would still match
                 outdated_symbols.append(hgnc_symbol)
-        else:
+        elif is_clinical:
```
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py:1003]: 👍

**Review by dnil — APPROVED** [review]

File: scout/server/blueprints/variants/controllers.py [hunk 2]
```diff
@@ -994,22 +994,30 @@ def check_form_gene_symbols(
 
         if hgnc_gene is None:
             not_found_symbols.append(hgnc_symbol)
+        elif is_clinical is False:  # research
+            updated_hgnc_symbols.append(hgnc_symbol)
```
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py:998]: 🥇

**Comment by northwestwitch** [comment]
Now it works and after the search the form retains the gene name of the gene in the form. Tested on stage with this case: https://scout-stage.scilifelab.se/cust000/15035-miptest/variants

**Comment by northwestwitch** [comment]
Oh no the coverage went down. Perhaps I can increase it a bit

File: scout/server/blueprints/variants/controllers.py [hunk 3]
```diff
@@ -994,15 +994,15 @@ def check_form_gene_symbols(
 
         if hgnc_gene is None:
             not_found_symbols.append(hgnc_symbol)
-        elif is_clinical is False:  # research
+        elif is_clinical is False:  # research variants
             updated_hgnc_symbols.append(hgnc_symbol)
-        elif is_clinical and hgnc_gene["hgnc_id"] in clinical_hgnc_ids:
+        elif hgnc_gene["hgnc_id"] in clinical_hgnc_ids:  # clinical variants
```
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py:999]: What if it is None then, eh, what if?
Sorry, looks good

Comment (northwestwitch) [reply scout/server/blueprints/variants/controllers.py:999]: I'm going to merge before I feel tempted to modify it again! 😆

**Pull request merged**